[BUG]: MCP RBAC + protocol e2e gates fail on the multi-replica stack

### Summary

The MCP protocol gates (`make test-mcp-rbac` and `make test-mcp-protocol-e2e`) fail on the
documented 3-replica `make testing-up` stack. The failure is deterministic (`test-mcp-rbac` ran
twice with identical results) and reproduces identically on v1.0.2 and v1.0.3.

### Steps to reproduce

1. Build the gateway image:

       make docker

2. Bring up the documented 3-replica testing stack:

       make testing-down compose-clean testing-up

3. Wait for the stack to be healthy (3 gateway replicas; `http://localhost:8080/health` returns 200).

4. Export the required env (`make` runs pytest via `uv run`, which does not load `.env`):

       export JWT_SECRET_KEY=$(grep -E '^JWT_SECRET_KEY=' .env | cut -d= -f2-)
       export PLATFORM_ADMIN_EMAIL=$(grep -E '^PLATFORM_ADMIN_EMAIL=' .env | cut -d= -f2-)
       export MCP_CLI_BASE_URL=http://localhost:8080

5. Run the gates:

       make test-mcp-rbac test-mcp-protocol-e2e

### Results (summary)

| Target | Runs | Passed | Failed / Errors | Skipped | Rerun |
|---|---|---|---|---|---|
| `test-mcp-rbac` | 2 (identical) | 13 | 27 failed | 0 | 23 |
| `test-mcp-protocol-e2e` | 1 | 2 | 19 errors | 1 | 7 |

Same result on v1.0.2 (`test-mcp-rbac`: 26 failed / 14 passed; `test-mcp-protocol-e2e`: 19 errors / 2 passed / 1 skipped), so this is pre-existing, not introduced in v1.0.3.

<details>
<summary>Full §5.4 failure report (counts + failing test list, Release/v1.0.3)</summary>

Stack: `make testing-up` (default `GATEWAY_REPLICAS=3`); all gateway replicas + nginx healthy; `/health` = 200; required env exported (`JWT_SECRET_KEY`, `PLATFORM_ADMIN_EMAIL`, `MCP_CLI_BASE_URL`).

| Target | Runs | Passed | Failed / Errors | Skipped | Rerun | Duration |
|---|---|---|---|---|---|---|
| `test-mcp-rbac` | 2 (identical) | 13 | 27 failed | 0 | 23 | ~340s |
| `test-mcp-protocol-e2e` | 1 | 2 | 19 errors | 1 | 7 | ~146s |

`test-mcp-rbac` was run twice and produced identical results both times (same counts, same failing tests), so the failure is deterministic, not flaky.

The dominant failure on both targets (every test that opens an MCP client session) occurs at the streamable HTTP `/mcp/` initialize handshake (`fastmcp/client/client.py:532`).

#### `test-mcp-rbac` — 27 failing (reproduced across both runs)

One distinct RBAC-visibility assertion; the other 26 are the MCP session-init failure.

Visibility assertion (separate failure, not session-init):
- `TestServerVisibilityViaAPI::test_admin_sees_public_and_team_via_http` — `AssertionError: PR #4341: admin via HTTP must NOT see private server (own-private collapsed by get_scoped_resource_access_context)`

MCP session init failures (26):
- `TestMcpToolsVisibilityByRole::test_admin_sees_all_tools`
- `TestMcpToolsVisibilityByRole::test_developer_sees_public_and_team_tools`
- `TestMcpToolsVisibilityByRole::test_viewer_sees_public_and_team_tools`
- `TestMcpToolsVisibilityByRole::test_outsider_sees_only_public_tools`
- `TestMcpToolsVisibilityByRole::test_team_admin_sees_public_and_team_tools`
- `TestMcpResourcesPromptsByRole::test_admin_resources`
- `TestMcpResourcesPromptsByRole::test_admin_prompts`
- `TestMcpResourcesPromptsByRole::test_developer_resources_and_prompts`
- `TestMcpResourcesPromptsByRole::test_outsider_resources_and_prompts`
- `TestMcpToolCallByRole::test_admin_calls_tool_success`
- `TestMcpToolCallByRole::test_developer_can_execute_on_default_endpoint`
- `TestMcpToolCallByRole::test_team_admin_can_execute_on_default_endpoint`
- `TestMcpToolCallByRole::test_viewer_can_execute_on_default_endpoint`
- `TestMcpScopedTokenPermissions::test_tools_read_only_token_can_initialize`
- `TestMcpScopedTokenPermissions::test_read_execute_token_can_initialize`
- `TestMcpScopedTokenPermissions::test_unscoped_admin_token_can_call_tools`
- `TestMcpSSETransport::test_sse_tools_discoverable`
- `TestMcpSSETransport::test_sse_get_system_time`
- `TestMcpSSETransport::test_sse_convert_time`
- `TestMcpSSETransport::test_sse_resources_discoverable`
- `TestMcpSSETransport::test_sse_prompts_discoverable`
- `TestMcpPerServerEndpoint::test_public_token_accesses_public_server`
- `TestMcpPerServerEndpoint::test_team_member_accesses_team_server`
- `TestDenyPaths::test_revoked_token_fails`
- `TestCrossTransportConsistency::test_get_system_time_both_transports`
- `TestCrossTransportConsistency::test_convert_time_both_transports`

#### `test-mcp-protocol-e2e` — 19 erroring

All 19 error in the `client` fixture setup:
- `TestConnectivity::test_ping`
- `TestConnectivity::test_initialize_reports_server_info`
- `TestConnectivity::test_server_capabilities_include_core_surfaces`
- `TestConnectivity::test_multiple_calls_in_one_session`
- `TestTools::test_tools_list_nonempty`
- `TestTools::test_tools_have_required_fields`
- `TestTools::test_tools_include_gateway_prefixed`
- `TestTools::test_tool_input_schemas_are_json_schema_objects`
- `TestDiscovery::test_resources_list`
- `TestDiscovery::test_resources_read_roundtrip`
- `TestDiscovery::test_prompts_list`
- `TestDiscovery::test_prompt_get_renders`
- `TestToolCalls::test_get_system_time`
- `TestToolCalls::test_echo`
- `TestToolCalls::test_convert_time`
- `TestToolCalls::test_get_stats`
- `TestToolCalls::test_schema_error_preserves_payload`
- `TestToolCalls::test_schema_success_validates_payload`
- `TestToolCalls::test_nonexistent_tool`

Skipped (1): `test_mcp_protocol_e2e.py:411` — Rust MCP public transport not active (expected; Rust mode off).
Passed (2): unauthenticated `/mcp/` → 401, and invalid method → 200 (the only checks that don't open an MCP session).

</details>

### Affected tests
- `tests/live_gateway/mcp/test_mcp_rbac_transport.py`
- `tests/live_gateway/mcp/test_mcp_protocol_e2e.py`

--- a/.secrets.baseline
+++ b/.secrets.baseline
@@ -3,7 +3,7 @@
     "files": "(?x)( package-lock\\.json$ |Cargo\\.lock$ |uv\\.lock$ |go\\.sum$ |mcpgateway/sri_hashes\\.json$ )|^.secrets.baseline$",
     "lines": null
   },
-  "generated_at": "2026-06-26T13:51:56Z",
+  "generated_at": "2026-06-26T12:18:07Z",
   "plugins_used": [
     {
       "name": "AWSKeyDetector"
@@ -232,7 +232,7 @@
         "hashed_secret": "f4793151b0607198d4de9b1ca458d3e25adf1cb7",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 121,
+        "line_number": 120,
         "type": "Secret Keyword",
         "verified_result": null
       },
@@ -240,7 +240,7 @@
         "hashed_secret": "fa9beb99e4029ad5a6615399e7bbae21356086b3",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 218,
+        "line_number": 217,
         "type": "Secret Keyword",
         "verified_result": null
       },
@@ -248,7 +248,7 @@
         "hashed_secret": "7b4455a56fbf1d198e45e04c437488514645a82c",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 220,
+        "line_number": 219,
         "type": "Secret Keyword",
         "verified_result": null
       },
@@ -256,7 +256,7 @@
         "hashed_secret": "90bd1b48e958257948487b90bee080ba5ed00caa",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 292,
+        "line_number": 291,
         "type": "Hex High Entropy String",
         "verified_result": null
       },
@@ -264,7 +264,7 @@
         "hashed_secret": "48ffbad96aa9c2b33f9486f5a3c2108198acb518",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 293,
+        "line_number": 292,
         "type": "Hex High Entropy String",
         "verified_result": null
       }
@@ -1306,7 +1306,7 @@
         "hashed_secret": "e204d28a2874f6123747650d3e4003d4357d75eb",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 855,
+        "line_number": 852,
         "type": "Secret Keyword",
         "verified_result": null
       },
@@ -1314,17 +1314,7 @@
         "hashed_secret": "fa9beb99e4029ad5a6615399e7bbae21356086b3",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 1150,
-        "type": "Secret Keyword",
-        "verified_result": null
-      }
-    ],
-    "docs/docs/architecture/oauth-design.md": [
-      {
-        "hashed_secret": "5c216af8faacfff02f8ee00326ecfcfb3334922f",
-        "is_secret": false,
-        "is_verified": false,
-        "line_number": 153,
+        "line_number": 1147,
         "type": "Secret Keyword",
         "verified_result": null
       }
@@ -2543,22 +2533,6 @@
         "line_number": 570,
         "type": "Secret Keyword",
         "verified_result": null
-      },
-      {
-        "hashed_secret": "4a0a2df96d4c9a13a282268cab33ac4b8cbb2c72",
-        "is_secret": false,
-        "is_verified": false,
-        "line_number": 643,
-        "type": "Secret Keyword",
-        "verified_result": null
-      },
-      {
-        "hashed_secret": "f4793151b0607198d4de9b1ca458d3e25adf1cb7",
-        "is_secret": false,
-        "is_verified": false,
-        "line_number": 645,
-        "type": "Secret Keyword",
-        "verified_result": null
       }
     ],
     "docs/docs/manage/oauth.md": [
@@ -2706,7 +2680,7 @@
         "hashed_secret": "1513802ada0ec04c2446206e16baa7256ebff74b",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 601,
+        "line_number": 598,
         "type": "Secret Keyword",
         "verified_result": null
       }
@@ -3077,26 +3051,10 @@
         "verified_result": null
       },
       {
-        "hashed_secret": "4a0a2df96d4c9a13a282268cab33ac4b8cbb2c72",
-        "is_secret": false,
-        "is_verified": false,
-        "line_number": 777,
-        "type": "Secret Keyword",
-        "verified_result": null
-      },
-      {
-        "hashed_secret": "5c216af8faacfff02f8ee00326ecfcfb3334922f",
-        "is_secret": false,
-        "is_verified": false,
-        "line_number": 799,
-        "type": "Secret Keyword",
-        "verified_result": null
-      },
-      {
         "hashed_secret": "157b314984dacdea10427c1ae3a856d286318e0d",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 902,
+        "line_number": 844,
         "type": "Secret Keyword",
         "verified_result": null
       }
@@ -3363,20 +3321,20 @@
         "verified_result": null
       }
     ],
-    "docs/docs/using/agents/beeai.md": [
+    "docs/docs/using/agents/bee.md": [
       {
         "hashed_secret": "5546721ffdfc2e5b0e4c0da38f10774f9ad50b09",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 35,
+        "line_number": 39,
         "type": "Secret Keyword",
         "verified_result": null
       },
       {
-        "hashed_secret": "ec545cfb40ff2f1b4eb959dbd0e3177236c540fa",
+        "hashed_secret": "be0e0678e7132e8f991d463818e1befbaf3aad9d",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 203,
+        "line_number": 61,
         "type": "Secret Keyword",
         "verified_result": null
       }

--- a/.secrets.baseline
+++ b/.secrets.baseline
@@ -3,7 +3,7 @@
     "files": "(?x)( package-lock\\.json$ |Cargo\\.lock$ |uv\\.lock$ |go\\.sum$ |mcpgateway/sri_hashes\\.json$ )|^.secrets.baseline$",
     "lines": null
   },
-  "generated_at": "2026-06-26T12:18:07Z",
+  "generated_at": "2026-06-26T14:48:00Z",
   "plugins_used": [
     {
       "name": "AWSKeyDetector"
@@ -232,7 +232,7 @@
         "hashed_secret": "f4793151b0607198d4de9b1ca458d3e25adf1cb7",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 120,
+        "line_number": 121,
         "type": "Secret Keyword",
         "verified_result": null
       },
@@ -240,7 +240,7 @@
         "hashed_secret": "fa9beb99e4029ad5a6615399e7bbae21356086b3",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 217,
+        "line_number": 218,
         "type": "Secret Keyword",
         "verified_result": null
       },
@@ -248,7 +248,7 @@
         "hashed_secret": "7b4455a56fbf1d198e45e04c437488514645a82c",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 219,
+        "line_number": 220,
         "type": "Secret Keyword",
         "verified_result": null
       },
@@ -256,7 +256,7 @@
         "hashed_secret": "90bd1b48e958257948487b90bee080ba5ed00caa",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 291,
+        "line_number": 292,
         "type": "Hex High Entropy String",
         "verified_result": null
       },
@@ -264,7 +264,7 @@
         "hashed_secret": "48ffbad96aa9c2b33f9486f5a3c2108198acb518",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 292,
+        "line_number": 293,
         "type": "Hex High Entropy String",
         "verified_result": null
       }
@@ -1306,7 +1306,7 @@
         "hashed_secret": "e204d28a2874f6123747650d3e4003d4357d75eb",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 852,
+        "line_number": 855,
         "type": "Secret Keyword",
         "verified_result": null
       },
@@ -1314,7 +1314,17 @@
         "hashed_secret": "fa9beb99e4029ad5a6615399e7bbae21356086b3",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 1147,
+        "line_number": 1150,
+        "type": "Secret Keyword",
+        "verified_result": null
+      }
+    ],
+    "docs/docs/architecture/oauth-design.md": [
+      {
+        "hashed_secret": "5c216af8faacfff02f8ee00326ecfcfb3334922f",
+        "is_secret": false,
+        "is_verified": false,
+        "line_number": 153,
         "type": "Secret Keyword",
         "verified_result": null
       }
@@ -2533,6 +2543,22 @@
         "line_number": 570,
         "type": "Secret Keyword",
         "verified_result": null
+      },
+      {
+        "hashed_secret": "4a0a2df96d4c9a13a282268cab33ac4b8cbb2c72",
+        "is_secret": false,
+        "is_verified": false,
+        "line_number": 643,
+        "type": "Secret Keyword",
+        "verified_result": null
+      },
+      {
+        "hashed_secret": "f4793151b0607198d4de9b1ca458d3e25adf1cb7",
+        "is_secret": false,
+        "is_verified": false,
+        "line_number": 645,
+        "type": "Secret Keyword",
+        "verified_result": null
       }
     ],
     "docs/docs/manage/oauth.md": [
@@ -2680,7 +2706,7 @@
         "hashed_secret": "1513802ada0ec04c2446206e16baa7256ebff74b",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 598,
+        "line_number": 601,
         "type": "Secret Keyword",
         "verified_result": null
       }
@@ -3051,10 +3077,26 @@
         "verified_result": null
       },
       {
+        "hashed_secret": "4a0a2df96d4c9a13a282268cab33ac4b8cbb2c72",
+        "is_secret": false,
+        "is_verified": false,
+        "line_number": 777,
+        "type": "Secret Keyword",
+        "verified_result": null
+      },
+      {
+        "hashed_secret": "5c216af8faacfff02f8ee00326ecfcfb3334922f",
+        "is_secret": false,
+        "is_verified": false,
+        "line_number": 799,
+        "type": "Secret Keyword",
+        "verified_result": null
+      },
+      {
         "hashed_secret": "157b314984dacdea10427c1ae3a856d286318e0d",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 844,
+        "line_number": 902,
         "type": "Secret Keyword",
         "verified_result": null
       }
@@ -3321,20 +3363,20 @@
         "verified_result": null
       }
     ],
-    "docs/docs/using/agents/bee.md": [
+    "docs/docs/using/agents/beeai.md": [
       {
         "hashed_secret": "5546721ffdfc2e5b0e4c0da38f10774f9ad50b09",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 39,
+        "line_number": 35,
         "type": "Secret Keyword",
         "verified_result": null
       },
       {
-        "hashed_secret": "be0e0678e7132e8f991d463818e1befbaf3aad9d",
+        "hashed_secret": "ec545cfb40ff2f1b4eb959dbd0e3177236c540fa",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 61,
+        "line_number": 203,
         "type": "Secret Keyword",
         "verified_result": null
       }

--- a/tests/live_gateway/helpers/mcp_test_helpers.py
+++ b/tests/live_gateway/helpers/mcp_test_helpers.py
@@ -36,7 +36,9 @@ import pytest
 # ---------------------------------------------------------------------------
 # Configuration
 # ---------------------------------------------------------------------------
-BASE_URL = os.getenv("MCP_CLI_BASE_URL", "http://127.0.0.1:8080")  # IPv4 explicit — `localhost` resolves IPv6-first under the test conftest's getaddrinfo stub, which docker-compose doesn't bind
+# Force IPv4: localhost can resolve to ::1 first, but compose only publishes the
+# gateway on IPv4, so an IPv6-first client hangs. Covers MCP_CLI_BASE_URL too.
+BASE_URL = os.getenv("MCP_CLI_BASE_URL", "http://127.0.0.1:8080").replace("//localhost", "//127.0.0.1")
 JWT_SECRET = os.getenv("JWT_SECRET_KEY", "my-test-key-but-now-longer-than-32-bytes")
 ADMIN_EMAIL = os.getenv("PLATFORM_ADMIN_EMAIL", "admin@example.com")
 TOKEN_EXPIRY = os.getenv("MCP_CLI_TOKEN_EXPIRY", "60")  # minutes

